compiler/p/codegen/ControlFlowEvaluator.cpp

```diff
@@ -3836,62 +3841,65 @@ static TR::Register *generateMaxMin(TR::Node *node, TR::CodeGenerator *cg, bool
          }
       if (two_reg)
          {
-         generateTrg1Src1Instruction(cg, move_op, node, trgReg->getLowOrder(), reg->getLowOrder());
-         generateTrg1Src1Instruction(cg, move_op, node, trgReg->getHighOrder(), reg->getHighOrder());
+         generateTrg1Src1Instruction(cg, move_op, node, trgReg->getLowOrder(), src1Reg->getLowOrder());
+         generateTrg1Src1Instruction(cg, move_op, node, trgReg->getHighOrder(), src1Reg->getHighOrder());
          }
       else
          {
-         generateTrg1Src1Instruction(cg, move_op, node, trgReg, reg);
+         generateTrg1Src1Instruction(cg, move_op, node, trgReg, src1Reg);
          }
       }
 
-   int n = node->getNumChildren();
-   for (int i = 1; i < n; i++)
+   TR::Register *condReg = cg->allocateRegister(TR_CCR);
+   TR::LabelSymbol *label = generateLabelSymbol(cg);
+   TR::LabelSymbol *nan_label;
+   if (check_nan)
+      nan_label = generateLabelSymbol(cg);
+   TR::RegisterDependencyConditions *dep;
+   if (two_reg)
       {
-      TR::Node *child = node->getChild(i);
-      TR::Register *reg = cg->evaluate(child);
-      TR::Register *condReg = cg->allocateRegister(TR_CCR);
-      TR::LabelSymbol *label = generateLabelSymbol(cg);
-      TR::RegisterDependencyConditions *dep;
-      if (two_reg)
+      TR::PPCControlFlowInstruction *cfop = (TR::PPCControlFlowInstruction *)
+         generateControlFlowInstruction(cg, TR::InstOpCode::iflong, node, NULL /*deps */);
+      cfop->addTargetRegister(condReg);
+      cfop->addSourceRegister(trgReg->getHighOrder());
+      cfop->addSourceRegister(trgReg->getLowOrder());
+      cfop->addSourceRegister(src2Reg->getHighOrder());
+      cfop->addSourceRegister(src2Reg->getLowOrder());
+      cfop->setLabelSymbol(label);
+      cfop->setOpCode2Value(max ? TR::InstOpCode::bge : TR::InstOpCode::ble);
+      generateTrg1Src1Instruction(cg, move_op, node, trgReg->getLowOrder(), src2Reg->getLowOrder());
+      generateTrg1Src1Instruction(cg, move_op, node, trgReg->getHighOrder(), src2Reg->getHighOrder());
+
+      dep = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 4, cg->trMemory());
+      dep->addPostCondition(trgReg->getLowOrder(), TR::RealRegister::NoReg);
+      dep->addPostCondition(trgReg->getHighOrder(), TR::RealRegister::NoReg);
+      dep->addPostCondition(src2Reg->getLowOrder(), TR::RealRegister::NoReg);
+      dep->addPostCondition(src2Reg->getHighOrder(), TR::RealRegister::NoReg);
```

Comment:
It seems that `condReg` is missing in these dependencies.

Reply:
Fixed in d04f5d9